remill/Arch/Runtime/Operators.h

```diff
@@ -498,16 +498,49 @@ MAKE_CMPXCHG(128, uint, 128)
 #undef MAKE_CMPXCHG
 #define UCmpXchg(op, oldval, newval) _CmpXchg(memory, op, oldval, newval)
 
+#define MAKE_ATOMIC_ACCESSOR(type_prefix, size) \
```

Comment:
I think you scan simplify a bunch of these various atomic operation-making macros into:
```c++
#define MAKE_ATOMIC_INTRINSIC(name, intrinsic_name, size, type_prefix, op)\
  template<typename T> \
  ALWAYS_INLINE type_prefix ## size ## _t _U ## name( \
      Memory *&memory, MnW<T> addr, type_prefix ## size ## _t &value) { \
    memory = intrinsic_name ## size(memory, addr.addr, value); \
    return value; \
  } \
  \
  template<typename T> \
  ALWAYS_INLINE type_prefix ## size ## _t _U ## name ( \
      Memory *&memory, RnW<T> addr, type_prefix ## size ## _t &value) { \
    auto prev_value = *reinterpret_cast<type_prefix ## size ## _t*>(addr.val_ref); \
    *addr.val_ref = prev_value op value; \
    value = prev_value; \
    return value; \
  }

#define MAKE_ATOMIC(name, op)  \
    MAKE_ATOMIC_INTRINSIC(name, 8, uint, op) \
    MAKE_ATOMIC_INTRINSIC(name, 16, uint, op) \
    MAKE_ATOMIC_INTRINSIC(name, 32, uint, op) \
    MAKE_ATOMIC_INTRINSIC(name, 64, uint, op) \

MAKE_ATOMIC(FetchAdd, +)
MAKE_ATOMIC(FetchSub, -)
MAKE_ATOMIC(FetchOr, |)
MAKE_ATOMIC(FetchAnd, &)
MAKE_ATOMIC(FetchXor, ^)

#undef MAKE_ATOMIC
#undef MAKE_ATOMIC_INTRINSIC
```

Reply:
updated with some fixes